homeassistant/components/overseerr/services.py

```diff
@@ -99,14 +143,164 @@ async def _async_get_requests(call: ServiceCall) -> ServiceResponse:
+        "results": cast(
+            list[JsonValueType], [asdict(result) for result in search_results]
+        )
+    }
+
+
+async def __request_media(
+    client: OverseerrClient,
+    media_type: MediaType,
+    tmdb_id: int,
+    seasons: int | list[int] | Literal['all'] | None = None,
+) -> Any:
+    """Request media in Overseerr."""
+    # Convert single integer to list for seasons
+    if isinstance(seasons, int):
+        seasons = [seasons]
+
+    try:
+        LOGGER.debug(
+            "Requesting %s with TMDB ID %s (seasons: %s)",
+            media_type,
+            tmdb_id,
+            seasons if seasons else "none",
+        )
+        request = await client.create_request(media_type, tmdb_id, seasons)
+    except OverseerrConnectionError as err:
+        LOGGER.error(
+            "Error requesting %s with TMDB ID %s: %s",
+            media_type,
+            tmdb_id,
+            str(err),
+        )
+        raise HomeAssistantError(
+            translation_domain=DOMAIN,
+            translation_key="connection_error",
+            translation_placeholders={"error": str(err)},
+        ) from err
+
+    return request
+
+
+async def _async_request_media(call: ServiceCall) -> ServiceResponse:
+    """Request media in Overseerr."""
+    entry = _async_get_entry(call.hass, call.data[ATTR_CONFIG_ENTRY_ID])
+    client = entry.runtime_data.client
+    media_type = MediaType(call.data[ATTR_MEDIA_TYPE])
+    tmdb_id = call.data[ATTR_TMDB_ID]
+    seasons = call.data.get(ATTR_SEASONS)
+
+    request = await __request_media(client, media_type, tmdb_id, seasons)
+
+    return {"request": cast(JsonValueType, asdict(request))}
+
+
+async def _async_search_and_request(call: ServiceCall) -> ServiceResponse:
+    """Search for media and request the first result in Overseerr."""
+    entry = _async_get_entry(call.hass, call.data[ATTR_CONFIG_ENTRY_ID])
+    client = entry.runtime_data.client
+    query = call.data[ATTR_QUERY]
+    seasons = call.data.get(ATTR_SEASONS)
+
+    search_results = await __search_media(client, query)
+
+    if not search_results:
+        LOGGER.error("No results found for query '%s'", query)
+        raise HomeAssistantError(
+            translation_domain=DOMAIN,
+            translation_key="no_results",
+            translation_placeholders={"query": query},
+        )
+
+    first_result = search_results[0]
+
+    media_type = first_result.media_type
+    tmdb_id = first_result.id
+
+    # Only use seasons parameter if it's a TV show
+    request_seasons = None
+    if media_type == "tv" and seasons:
+        request_seasons = seasons
+
+    request = await __request_media(client, media_type, tmdb_id, request_seasons)
+
+    return {
+        "request": cast(JsonValueType, asdict(request)),
+        "media": {
+            "type": media_type,
+            "id": tmdb_id,
+            "title": getattr(
+                first_result, "title", getattr(first_result, "name", "Unknown")
+            ),
```

Comment:
This is very hard to read.

```suggestion
             "title": first_result.title if hasattr(first_result, "title") else getattr(first_result, "name", "Unknown")
```